### ISSUE TITLE
Hierarchy Motion Vector Mode Override Component

### ISSUE THREAD

**Issue by BaxterOttoman** [Yellow-Dog-Man/Resonite-Issues#933]
### Is your feature request related to a problem? Please describe.

I would like to suggest a component that serves to mark a hierarchy to override the motion vector mode on renderers found within that hierarchy.

### Describe the solution you'd like

The use I imagine is you would apply the component to a slot, similarly to how you apply the Hidden Layer component, and renderers in children to that slow would have their motion vector mode to be overridden to the mode selected in the override component.

### Describe alternatives you've considered

Marking manually works, but this could allow things to be dynamically added or removed from a hierarchy without having to manually update how they're set to visualize their blur.

**Comment by 5H4D0W-X** [comment]
This could also be combined with overrides for other render settings such as shadow casting mode

**Comment by Frooxius** [comment]
I don't think this would work well as a "blanket component". Generally you should be able to override motion vectors on particular renderers as needed.

If there was a component that marks this for the entire hierarchy, this would likely add a global performance overhead for everything - every single renderer component now has to keep checking if it's under such override component or not. We can amortize this cost, but it would be at the cost of using more memory (and little bit of compute) for every single slot, everywhere, regardless of whether you're using this component or not - which in most cases, you won't be.

That's a really big performance impact to ask, so there would need to be really big reason for doing it this way.

The other alternative is we put all the cost into the component itself, but that will make it a lot more expensive, as it'll have to more actively track all the renderers in its hierarchy - those can be added, removed, moved and so on.

Unless you have some very significant reason for needing this component, we're probably not going to add it.

**Comment by Nytra** [comment]
Perhaps this could be added as some kind of tool instead. The MeshVisibilityTool is able to do something like this for toggling MeshRenderer enabled state and also their shadows. It could become a new feature of that tool maybe.

**Comment by Frooxius** [comment]
Yeah if you want a tool to mass set it, that's significantly easier. It wouldn't automatically track any new additions/removals for it. But it's a very different purpose from a component, so I'm not sure if that would satisfy the original request.

We can of course add that independent of that too if there's interest in that kind of tool.
